### ISSUE TITLE
mac running MoonDream Node node reporting errors

### ISSUE THREAD

**Issue by yiwangsimple** [gokayfem/ComfyUI_VLM_nodes#21]
Error occurred when executing MoonDream:

The expanded size of the tensor (745) must match the existing size (746) at non-singleton dimension 1. Target sizes: [1, 745]. Tensor sizes: [1, 746]

**Comment by gokayfem** [comment]
can you send the full error please, its really hard to debug if i dont know the source

**Comment by yiwangsimple** [comment]
Okay, please let me try again.

**Comment by yiwangsimple** [comment]
> can you send the full error please, its really hard to debug if i dont know the source
ERROR:root:!!! Exception during processing !!!
ERROR:root:Traceback (most recent call last):
  File "/Users/weiwei/ComfyUI/execution.py", line 152, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
  File "/Users/weiwei/ComfyUI/execution.py", line 82, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
  File "/Users/weiwei/ComfyUI/execution.py", line 75, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream_script.py", line 76, in answer_questions
    full_sentence = self.text_model.answer_question(image_embeds, question)
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/text_model.py", line 79, in answer_question
    answer = self.generate(
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/text_model.py", line 71, in generate
    output_ids = self.model.generate(
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/transformers/generation/utils.py", line 1544, in generate
    return self.greedy_search(
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/transformers/generation/utils.py", line 2404, in greedy_search
    outputs = self(
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 992, in forward
    hidden_states = self.transformer(
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 933, in forward
    hidden_states = layer(
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 734, in forward
    attn_outputs = self.mixer(
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 688, in forward
    attn_output = self._forward_cross_attn(
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 664, in _forward_cross_attn
    return self.inner_cross_attn(
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/amp/autocast_mode.py", line 16, in decorate_autocast
    return func(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/amp/autocast_mode.py", line 16, in decorate_autocast
    return func(*args, **kwargs)
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 462, in forward
    padding_mask.masked_fill_(key_padding_mask, 0.0)
RuntimeError: The expanded size of the tensor (749) must match the existing size (750) at non-singleton dimension 1.  Target sizes: [1, 749].  Tensor sizes: [1, 750]

**Comment by yiwangsimple** [comment]
> can you send the full error please, its really hard to debug if i dont know the source

ERROR:root:Traceback (most recent call last):
  File "/Users/weiwei/ComfyUI/execution.py", line 152, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
  File "/Users/weiwei/ComfyUI/execution.py", line 82, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
  File "/Users/weiwei/ComfyUI/execution.py", line 75, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream_script.py", line 76, in answer_questions
    full_sentence = self.text_model.answer_question(image_embeds, question)
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/text_model.py", line 79, in answer_question
    answer = self.generate(
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/text_model.py", line 71, in generate
    output_ids = self.model.generate(
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/transformers/generation/utils.py", line 1544, in generate
    return self.greedy_search(
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/transformers/generation/utils.py", line 2404, in greedy_search
    outputs = self(
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 992, in forward
    hidden_states = self.transformer(
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 933, in forward
    hidden_states = layer(
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 734, in forward
    attn_outputs = self.mixer(
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 688, in forward
    attn_output = self._forward_cross_attn(
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 664, in _forward_cross_attn
    return self.inner_cross_attn(
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/amp/autocast_mode.py", line 16, in decorate_autocast
    return func(*args, **kwargs)
  File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/amp/autocast_mode.py", line 16, in decorate_autocast
    return func(*args, **kwargs)
  File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 462, in forward
    padding_mask.masked_fill_(key_padding_mask, 0.0)
RuntimeError: The expanded size of the tensor (749) must match the existing size (750) at non-singleton dimension 1.  Target sizes: [1, 749].  Tensor sizes: [1, 750]

Prompt executed in 3695.42 seconds

**Comment by yiwangsimple** [comment]
Error occurred when executing MoonDream:

The expanded size of the tensor (749) must match the existing size (750) at non-singleton dimension 1. Target sizes: [1, 749]. Tensor sizes: [1, 750]

File "/Users/weiwei/ComfyUI/execution.py", line 152, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
File "/Users/weiwei/ComfyUI/execution.py", line 82, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
File "/Users/weiwei/ComfyUI/execution.py", line 75, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream_script.py", line 76, in answer_questions
full_sentence = self.text_model.answer_question(image_embeds, question)
File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/text_model.py", line 79, in answer_question
answer = self.generate(
File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/text_model.py", line 71, in generate
output_ids = self.model.generate(
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/transformers/generation/utils.py", line 1544, in generate
return self.greedy_search(
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/transformers/generation/utils.py", line 2404, in greedy_search
outputs = self(
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
return self._call_impl(*args, **kwargs)
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
return forward_call(*args, **kwargs)
File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 992, in forward
hidden_states = self.transformer(
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
return self._call_impl(*args, **kwargs)
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
return forward_call(*args, **kwargs)
File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 933, in forward
hidden_states = layer(
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
return self._call_impl(*args, **kwargs)
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
return forward_call(*args, **kwargs)
File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 734, in forward
attn_outputs = self.mixer(
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
return self._call_impl(*args, **kwargs)
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
return forward_call(*args, **kwargs)
File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 688, in forward
attn_output = self._forward_cross_attn(
File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 664, in _forward_cross_attn
return self.inner_cross_attn(
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
return self._call_impl(*args, **kwargs)
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
return forward_call(*args, **kwargs)
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/amp/autocast_mode.py", line 16, in decorate_autocast
return func(*args, **kwargs)
File "/Users/weiwei/Envs/comfyui/lib/python3.10/site-packages/torch/amp/autocast_mode.py", line 16, in decorate_autocast
return func(*args, **kwargs)
File "/Users/weiwei/ComfyUI/custom_nodes/ComfyUI_VLM_nodes-mac/nodes/moondream/phi/modeling_phi.py", line 462, in forward
padding_mask.masked_fill_(key_padding_mask, 0.0)

**Comment by gokayfem** [comment]
i cant really iterate on this for now, i just released new model called UForm-Gen2 Qwen, you can try it, it also fast like moondream

**Comment by yiwangsimple** [comment]
> i cant really iterate on this for now, since i dont have a macbook. i just released new model called UForm-Gen2 Qwen, you can try it, it also fast like moondream

Okay, thanks.

**Comment by yiwangsimple** [comment]
> i cant really iterate on this for now, since i dont have a macbook. i just released new model called UForm-Gen2 Qwen, you can try it, it also fast like moondream

OK，Let me have a try, thanks !

**Comment by gokayfem** [comment]
this is fixed
